engine/classes/Elgg/Database/AdminNotices.php

```diff
@@ -61,27 +65,26 @@ function add($id, $message) {
 	 *
 	 * @return bool
 	 */
-	function delete($id) {
-		if (!$id) {
-			return false;
-		}
-		
+	public function delete($id = '') {
```

Comment:
why no quick return if empty $id?? if you need a delete all just create a new function

Reply:
Metadata can be empty

Reply:
when? and why would you want that usecase in this function? if there is no `id` value than you can not delete by `id`...

Reply:
It's an object, you can change it's metadata to whatever you please. Empty string is valid metadata and valid id.

Reply:
i understand that it all is valid, but it is not what the function is for... creation with an empty id is not allowed, so it makes logical sense that the deletion can not happen without it... was there a specific use case why you needed this change?

Reply:
It doesn't make sense to require an attribute and then check if attribute is empty, when we can just allow an empty attribute.

Reply:
Even scruitinizer thinks so https://scrutinizer-ci.com/g/Elgg/Elgg/inspections/58b5ea3b-2cb2-45c5-9991-4f61b65421fd

Reply:
could you add a test for `elgg_delete_admin_notice('')`?

Reply:
No, I am done with this PR. Take it or leave it

Reply:
When I ask you to add tests, you do this: https://github.com/Elgg/Elgg/issues/11528

Reply:
i thought you were the one that wanted all the tests... i think this is a rare case that gets easily broken, so a test would be good

Reply:
> When I ask you to add tests, you do this: #11528

and then it gets merged :)